seqr/views/utils/dataset_utils.py

```diff
@@ -396,26 +407,27 @@ def _load_rna_seq(model_cls, file_path, user, mapping_file, ignore_extra_samples
     samples, remaining_sample_ids = _match_and_update_rna_samples(
         projects=get_internal_projects(),
         user=user,
-        sample_ids=samples_by_id.keys(),
+        sample_project_tuples=samples_by_id.keys(),
         data_source=data_source,
         sample_id_to_individual_id_mapping=sample_id_to_individual_id_mapping,
-        raise_unmatched_error_template=None if ignore_extra_samples else 'Unable to find matches for the following samples: {sample_ids}'
+        raise_unmatched_error_template=None if ignore_extra_samples else 'Unable to find matches for the following samples: {sample_ids}',
     )
 
     warnings = []
     if validate_samples:
         samples = validate_samples(samples, sample_id_to_tissue_type, warnings)
 
     # Delete old data
-    individual_db_ids = {s.individual_id for s in samples}
-    to_delete = model_cls.objects.filter(sample__individual_id__in=individual_db_ids).exclude(sample__data_source=data_source)
+    to_delete = model_cls.objects.filter(sample__in=samples).exclude(sample__data_source=data_source)
     prev_loaded_individual_ids = set(to_delete.values_list('sample__individual_id', flat=True))
     if to_delete:
         model_cls.bulk_delete(user, to_delete)
 
     loaded_sample_ids = set(model_cls.objects.filter(sample__in=samples).values_list('sample_id', flat=True).distinct())
+    samples = Sample.objects.filter(id__in={s.id for s in samples} - loaded_sample_ids).annotate(
+        project_name=F('individual__family__project__name') if PROJECT_COL in expected_columns else Value(None, output_field=TextField()))
```

Comment:
We should be able to check `rnaseqtpm_set` or `rnaseqoutlier_set` or similar fields for samples requiring a loading without `loaded_sample_ids`. But I need to learn how to make it work.